### PR TITLE
Cluster Domain integration tests

### PR DESCRIPTION
**Pull request by adityadani**
**What type of PR is this?**
>integration-test

**What this PR does / why we need it**:
Integration tests for cluster domains

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
No

**Does this change need to be cherry-picked to a release branch?**:
No

### CODE REVIEW

**Comment by adityadani** [comment]
Will update the Gopkg.toml, once the sched-ops PR gets merged.

File: test/integration_test/clusterdomain_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	_, err := k8s.Instance().CreateClusterDomainUpdate(&v1alpha1.ClusterDomainUpdate{
+		ObjectMeta: meta.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.ClusterDomainUpdateSpec{
+			ClusterDomain: domain,
+			Active:        active,
+		},
+	})
```
Comment (disrani-px) [inline test/integration_test/clusterdomain_test.go]: Need a VaildateClusterDomainUpdate here to make sure the update was successful

Comment (disrani-px) [reply test/integration_test/clusterdomain_test.go]: The validate on the ClusterDomainStatus will also check that it's in the correct state but good to the check that status here too.

File: test/integration_test/clusterdomain_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+	// validate the following
+	// - migration is successful
+	// - app starts on cluster 1
+	validateAndDestroyMigration(t, ctxs, preMigrationCtx, true, false, false, true)
+
+	testClusterDomainsFailover(t, preMigrationCtx, ctxs)
+
+	testClusterDomainsFailback(t, preMigrationCtx, ctxs)
+
```
Comment (disrani-px) [inline test/integration_test/clusterdomain_test.go]: Need to destroy the app on the source cluster at the end of the test.

File: test/integration_test/clusterdomain_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+func testClusterDomainsFailback(
+	t *testing.T,
+	preMigrationCtx *scheduler.Context,
+	ctxs []*scheduler.Context,
+) {
+	// Failback the application
+
+	// destroy the app on cluster 2
+	err := schedulerDriver.Destroy(preMigrationCtx, nil)
```
Comment (disrani-px) [inline test/integration_test/clusterdomain_test.go]: destroyAndWait() does these both steps, you can use that if you want.

File: test/integration_test/clusterdomain_test.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+	// Reduce the replicas on cluster 1
+	scaleFactor, err := schedulerDriver.GetScaleFactorMap(ctxs[0])
+	require.NoError(t, err, "Unexpected error on GetScaleFactorMap")
+	for k, _ := range scaleFactor {
+		scaleFactor[k] = 1
+	}
+	schedulerDriver.ScaleApplication(ctxs[0], scaleFactor)
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state on remote cluster after migration")
```
Comment (disrani-px) [inline test/integration_test/clusterdomain_test.go]: `on source cluster`

File: test/integration_test/clusterdomain_test.go (hunk 2 above)
Comment (adityadani) [reply test/integration_test/clusterdomain_test.go]: It destroys the PVC as well.

File: test/integration_test/clusterdomain_test.go (hunk 3 above)
Comment (adityadani) [reply test/integration_test/clusterdomain_test.go]: It also deletes the PVC.

File: test/integration_test/clusterdomain_test.go (hunk 2 above)
Comment (adityadani) [reply test/integration_test/clusterdomain_test.go]: done

File: test/integration_test/clusterdomain_test.go (hunk 1 above)
Comment (adityadani) [reply test/integration_test/clusterdomain_test.go]: done

File: vendor/github.com/portworx/sched-ops/k8s/k8s.go [hunk 5]
```diff
@@ -515,6 +515,8 @@ type ClusterDomainsOps interface {
 	DeleteClusterDomainsStatus(string) error
 	// ListClusterDomainStatuses lists ClusterDomainsStatus
 	ListClusterDomainStatuses() (*v1alpha1.ClusterDomainsStatusList, error)
+	// ValidateClusterDomainsStatus validates the ClusterDomainsStatus
```
Comment (disrani-px) [inline vendor/github.com/portworx/sched-ops/k8s/k8s.go]: `Gopkg.lock` changes are missing?

**Pull request merged**